config/jobs/kubernetes-csi/volume-data-source-validator/volume-data-source-validator-config.yaml

```diff
@@ -24,11 +25,355 @@ presubmits:
         args:
         - ./.prow.sh
         env:
+        # We pick some version for which there are pre-built images for kind.
+        # Update only when the newer version is known to not cause issues,
+        # otherwise presubmit jobs may start to fail for reasons that are
+        # unrelated to the PR. Testing against the latest Kubernetes is covered
+        # by periodic jobs (see https://k8s-testgrid.appspot.com/sig-storage-csi-ci#Summary).
+        - name: CSI_PROW_KUBERNETES_VERSION
+          value: "1.21.0"
+        - name: CSI_PROW_USE_BAZEL
+          value: "false"
+        - name: CSI_PROW_KUBERNETES_DEPLOYMENT
+          value: "1.21"
+        - name: CSI_PROW_DEPLOYMENT_SUFFIX
+          value: ""
+        - name: CSI_PROW_DRIVER_VERSION
+          value: "v1.7.2"
         - name: CSI_SNAPSHOTTER_VERSION
-          value: "v3.0.3"
+          value: "v4.0.0"
+        - name: CSI_PROW_TESTS
+          value: "sanity serial parallel"
         # docker-in-docker needs privileged mode
         securityContext:
           privileged: true
         resources:
           requests:
             cpu: 2000m
+  - name: pull-kubernetes-csi-volume-data-source-validator-1-21-on-kubernetes-master
+    # Explicitly needs to be started with /test.
+    # This cannot be enabled by default because there's always the risk
+    # that something changes in master which breaks the pre-merge check.
+    always_run: false
+    optional: true
+    decorate: true
+    skip_report: false
+    labels:
+      preset-service-account: "true"
+      preset-dind-enabled: "true"
+      preset-bazel-remote-cache-enabled: "true"
+      preset-kind-volume-mounts: "true"
+    annotations:
+      testgrid-dashboards: sig-storage-csi-volume-data-source-validator
+      testgrid-tab-name: 1-21-on-kubernetes-master
+      description: Kubernetes-CSI pull job in repo volume-data-source-validator for non-alpha tests, using deployment 1.21 on Kubernetes master
```

Comment:
I might not understand correctly, but is it trying to run e2e test on k8s 1.21 cluster where AnyVolumeDataSource feature is disabled by default? Then, is the feature enabled somewhere?

Also, I'm wondering if the change introduced in https://kubernetes.io/blog/2021/08/30/volume-populators-redesigned/, or using `dataSourceRef` field, can be handled properly in the e2e test (I think that the tests need some tricky codes to handle such a case to achieve it).

It may be easier to start adding the job only for 1.24 first and consider adding more later.

Reply:
This PR is to enable the k8s e2e test runs in the repo in general. The test that needs AnyVolumeDataSource enabled should only run on the k8s version that supports the feature.

Reply:
Sorry for the delay, I'll address this on Monday.